src/liballoc/collections/btree/search.rs

```diff
@@ -32,6 +36,12 @@ where
     }
 }
 
+/// Looks up a given key in a given node, without recursion.
+/// Returns a `Found` with the handle of the matching KV, if any. Otherwise,
+/// returns a `GoDown` with the handle of the edge where the key might be found.
+/// If the node is a leaf, a `GoDown` edge is not an actual edge but a possible edge;
+/// in other words, it is the location at which to insert the key, or irrelevant if
+/// `BorrowType` is `marker::Immut`.
```

Comment:
Hm, I imagine it's not *irrelevant* if the type is immut -- it's still that edge, right? e.g. a hypothetical `[T]::binary_search`-like API on BTree would want that index/edge.

Reply:
I don't understand. It's all in the "node is a leaf" case, why would you care if you won't insert?

Reply:
Well, maybe it's not my business what someone wants to do with it, but it feels wrong to suggest it's the place to insert if it's an immutable handle.

Reply:
There's (eventual) plans to add something like the LinkedList Cursor API to BTreeMap so that you could actually navigate in the conceptual "Array" that the map represents. (In fact, the linked list cursors are considered as a proving ground for that API design).

As another example -- though also an API that I believe BTree's don't provide today -- say I have a non-existent needle x, and I want to find the predecessor/successor of that x without traversing O(n) elements (as the iterator API would force me to). With this index, I could return that information fairly easily with O(log(n)) accesses, right? (where n is the length/size of the map).

Reply:
And I didn't want to say if "marker::Mut", because I guess you can use it for "marker::Owned", but I'm not sure.

Reply:
It's the way binary_search on slices describes it, and this is the same API (I believe) so I think it's reasonable to do so here too.

> If the value is found then Result::Ok is returned, containing the index of the matching element. If there are multiple matches, then any one of the matches could be returned. If the value is not found then Result::Err is returned, containing the index where a matching element could be inserted while maintaining sorted order.

Reply:
You're absolutely right. How about "belongs"?

Reply:
I didn't see that comment. Yes, "could be inserted" sounds good, and "place to insert" also doesn't say the handle can be used to insert, it's just a location.

Reply:
Anyway, going offline now.

Reply:
(The quote was from the binary search docs, to be clear).

Could be inserted is fine, or some other wording. Anything works :)